cmd/livepeer/livepeer.go

```diff
@@ -385,6 +388,18 @@ func main() {
 			// TODO: Initialize Sender with an implementation
 			// of RoundsManager that reads from a cache
 			n.Sender = pm.NewSender(n.Eth, n.Eth)
+
+			if *pixelsPerUnit <= 0 {
+				// Can't divide by 0
+				glog.Fatalf("The amount of pixels per unit must be greater than 0, provided %d instead\n", *pixelsPerUnit)
```

Comment:
Good point, I did not realize that. It is indeed equivalent to `os.Exit(n)`

Reply:
Entirely correct. I should replace fatals for orchestrator setting basePrice as well then probably?

Reply:
Now in `./test_args.sh` our exit code checking statements don't work anymore, e.g. : 

```
# Orchestrator needs to explicitly set PricePerUnit otherwise it will default to 0 resulting in a fatal error
res=0
./livepeer -orchestrator -serviceAddr 127.0.0.1:8935 -transcoder -network rinkeby $ETH_ARGS || res=$?
[ $res -ne 0 ]
```
Checking for $res -eq 0 seems trivial if we are not checking the actual error being logged. 

It seems like `panic` actually executes the deferred functions normally 

> Panic is a built-in function that stops the ordinary flow of control and begins panicking. When the function F calls panic, execution of F stops, any deferred functions in F are executed normally, and then F returns to its caller.

So I'm using that for now, please let me know if that's not a good idea.

Reply:
While `panic` does get the job done here, I'm not a big fan of the output displayed to the user:

Ex.

```
panic: Price per unit of pixels must be greater than 0, provided -5 instead


goroutine 1 [running]:
main.main()
	/Users/yondonfu/Development/go/src/github.com/livepeer/go-livepeer/cmd/livepeer/livepeer.go:329 +0x559f
```

IMO the goroutine stack trace makes it seem like something went horribly wrong in the program execution in a similar manner as panics due to nil pointer references when in reality the source of the issue is user input as opposed to programmer error (anecdotally I am not accustomed to seeing this type of error format displayed when running CLI based programs).

An alternative could be to wrap everything that is currently in the `main()` function in its own helper function which is then invoked by the `main()` function. The `main()` function could check the error returned by the helper function and if it is non-nil, the `main()` function will call `os.Exit(1)`. 

Ex.

```
func lpMain() error {
    ...
}

func main() {
    if err := lpMain(); err != nil {
        os.Exit(1)
    }
}
```

Reply:
The current construction of the main function is something I wanted to discuss after researching good packet/code structure. In short all startup of services should be wrapped in helper methods so the clutter within the main itself is rather small. 

I'm just not sure if it would be ideal to include such a refactor into this PR or if it warrants a seperate issue down the line (optimization). 

Perhaps we should compromise here for the time being and either go with `panic` (testable based on error code),  `return (status code 0)` (not testable on error code, but perhaps in a different way I'm not familiar with), or wrapping the entire current `main` body as you decribe.

Reply:
Just on a sidenote coming back to the original use of `glog.Fatal`,  we are also using `glog.Fatal()` for many other checks such as setting service URI, auth webhook URL, etc.

So I will be updating those as well to use `glog.Errorf + return`

Reply:
Yeah I agree that we can keep the refactor of the `main()` function out of this PR.

I'm ok with using `panic` for now as wrapping the current body of `main()` will require a bunch of changes to return proper error values, but let's make sure to come back and address this particular issue in a separate PR.

Reply:
> Just on a sidenote coming back to the original use of glog.Fatal, we are also using glog.Fatal() for many other checks such as setting service URI, auth webhook URL, etc.

Coming back to my previous comment. I should update these to use panic as well then for proper execution of deferred functions.

Reply:
If we're planning on switching away from using `panic` in a separate PR, I think its ok to leave the existing usage of `glog.Fatal` for now just as long as we don't introduce any additional usage.